sdgym/synthesizers/base.py

```diff
@@ -54,8 +53,13 @@ def get_trained_synthesizer(self, data, metadata):
             obj:
                 The synthesizer object.
         """
-        metadata_class = MultiTableMetadata() if 'tables' in metadata else SingleTableMetadata()
-        metadata = metadata_class.load_from_dict(metadata)
+        metadata_object = Metadata()
+        if 'tables' not in metadata:
+            table_name = metadata_object._get_single_table_name()
+            metadata = metadata_object.load_from_dict(metadata, table_name)
```

Comment:
I don't think you need to pass the table name. It seems like it will handle the case where tables are missing and use the default name on its own 
https://github.com/sdv-dev/SDV/blob/363d8bdca29fa26c3626d86bb535a7ca656a8671/sdv/metadata/metadata.py#L123-L135